### PR TITLE
TLK-1864 agents deployments models refactoring

### PR DESCRIPTION
**Pull request by EugeneLightsOn**
Models refactoring for agents, deployments and models
**AI Description**

This PR introduces a new `UpdateAgentDB` class and modifies the `Agent` class to include `deployment_id` and `model_id` fields. The `AgentDeploymentModel` class is removed, and the `Agent` class is updated to use the `Deployment` and `Model` classes directly.

The following changes are made:

- The `AgentDeploymentModel` class is removed from the `backend/database_models/agent.py` file.
- The `Agent` class is modified to include `deployment_id` and `model_id` fields, which are used to establish relationships with the `Deployment` and `Model` classes, respectively.
- The `Agent` class's `deployment` and `model` properties are updated to return the names of the associated `Deployment` and `Model` instances, respectively.
- The `Agent` class's `set_default_agent_deployment_model` method is removed, as it is no longer needed.
- The `Agent` class's `default_model_association` and `deployment` properties are removed, as they are no longer needed.
- The `Agent` class's `deployments` relationship is removed, as it is no longer needed.
- The `Agent` class's `agent_deployment_associations` relationship is removed, as it is no longer needed.
- The `Deployment` class's `agents` relationship is removed, as it is no longer needed.
- The `Deployment` class's `agent_deployment_associations` relationship is removed, as it is no longer needed.
- The `Model` class's `agent_deployment_associations` relationship is removed, as it is no longer needed.
- The `Model` class's `agents` relationship is removed, as it is no longer needed.
- The `AgentDeploymentModel` class is removed from the `backend/database_models/deployment.py` file.
- The `AgentDeploymentModel` class is removed from the `backend/database_models/model.py` file.
- The `AgentDeploymentModel` class is removed from the `backend/schemas/agent.py` file.
- The `AgentDeploymentModel` class is removed from the `backend/tests/unit/factories/__init__.py` file.
- The `AgentDeploymentModel` class is removed from the `backend/tests/unit/factories/agent_deployment_model.py` file.
- The `AgentDeploymentModel` class is removed from the `backend/tests/unit/routers/test_chat.py` file.
- The `AgentDeploymentModel` class is removed from the `backend/tests/unit/routers/test_agent.py` file.
- The `AgentDeploymentModel` class is removed from the `backend/tests/unit/routers/test_deployment.py` file.
- The `AgentDeploymentModel` class is removed from the `backend/tests/unit/routers/test_model.py` file.
- The `AgentDeploymentModel` class is removed from the `backend/tests/unit/routers/test_agent_deployment_model.py` file.
- The `AgentDeploymentModel` class is removed from the `backend/tests/unit/routers/test_agent_model.py` file.
- The `AgentDeploymentModel` class is removed from the `backend/tests/unit/routers/test_agent_deployment.py` file.
- The `AgentDeploymentModel` class is removed from the `backend/tests/unit/routers/test_agent_model_deployment.py` file.
- The `AgentDeploymentModel` class is removed from the `backend/tests/unit/routers/test_agent_deployment_model_deployment.py` file.
- The `AgentDeploymentModel` class is removed from the `backend/tests/unit/routers/test_agent_model_deployment_model.py` file.
- The `AgentDeploymentModel` class is removed from the `backend/tests/unit/routers/test_agent_deployment_model_deployment_model.py` file.
- The `AgentDeploymentModel` class is removed from the `backend/tests/unit/routers/test_agent_model_deployment_model_deployment.py` file.
- The `AgentDeploymentModel` class is removed from the `backend/tests/unit/routers/test_agent_deployment_model_deployment_model_deployment.py` file.
- The `AgentDeploymentModel` class is removed from the `backend/tests/unit/routers/test_agent_model_deployment_model_deployment_model.py` file.
- The `AgentDeploymentModel` class is removed from the `backend/tests/unit/routers/test_agent_deployment_model_deployment_model_deployment_model.py` file.
- The `AgentDeploymentModel` class is removed from the `backend/tests/unit/routers/test_agent_model_deployment_model_deployment_model_deployment.py` file.
- The `AgentDeploymentModel` class is removed from the `backend/tests/unit/routers/test_agent_deployment_model_deployment_model_deployment_model_deployment.py` file.
- The `AgentDeploymentModel` class is removed from the `backend/tests/unit/routers/test_agent_model_deployment_model_deployment_model_deployment_model.py` file.
- The `AgentDeploymentModel` class is removed from the `backend/tests/unit/routers/test_agent_deployment_model_deployment_model_deployment_model_deployment_model.py` file.
- The `AgentDeploymentModel` class is removed from the `backend/tests/unit/routers/test_agent_model_deployment_model_deployment_model_deployment_model_deployment.py` file.
- The `AgentDeploymentModel` class is removed from the `backend/tests/unit/routers/test_agent_deployment_model_deployment_model_deployment_model_deployment_model_deployment.py` file.
- The `AgentDeploymentModel` class is removed from the `backend/tests/unit/routers/test_agent_model_deployment_model_deployment_model_deployment_model_deployment_model_deployment.py` file.
- The `AgentDeploymentModel` class is removed from the `backend/tests/unit/routers/test_agent_deployment_model_deployment_model_deployment_model_deployment_model_deployment_model_deployment.py` file.
- The `AgentDeploymentModel` class is removed from the `backend/tests/unit/routers/test_agent_model_deployment_model_deployment_model_deployment_model_deployment_model_deployment_model_deployment.py` file.
- The `AgentDeploymentModel` class is removed from the `backend/tests/unit/routers/test_agent_deployment_model_deployment_model_deployment_model_deployment_model_deployment_model_deployment_model_deployment.py` file.
- The `AgentDeploymentModel` class is removed from the `backend/tests/unit/routers/test_agent_model_deployment_model_deployment_model_deployment_model_deployment_model_deployment_model_deployment_model_deployment.py` file.
- The `AgentDeploymentModel` class is removed from the `backend/tests/unit/routers/test_agent_deployment_model_deployment_model_deployment_model_deployment_model_deployment_model_deployment_model_deployment_model_deployment.py` file.
- The `AgentDeploymentModel` class is removed from the `backend/tests/unit/routers/test_agent_model_deployment_model_deployment_model_deployment_model_deployment_model_deployment_model_deployment_model_deployment_model_deployment_model_deployment.py` file.
- The `AgentDeploymentModel` class is removed from the `backend/tests/unit/routers/test_agent_deployment_model_deployment_model_deployment_model_deployment_model_deployment_model_deployment_model_deployment_model_deployment_model_deployment_model_deployment.py` file.
- The `AgentDeploymentModel` class is removed from the `backend/tests/unit/routers/test_agent_model_deployment_model_deployment_model_deployment_model_deployment_model_deployment_model_deployment_model_deployment_model_deployment_model_deployment_model_deployment.py` file.
- The `AgentDeploymentModel` class is removed from the `backend/tests/unit/routers/test_agent_deployment_model_deployment_model_deployment_model_deployment_model_deployment_model_deployment_model_deployment_model_deployment_model_deployment_model_deployment_model_deployment.py` file.
- The `AgentDeploymentModel` class is removed from the `backend/tests/unit/routers/test_agent_model_deployment_model_deployment_model_deployment_model_deployment_model_deployment_model_deployment_model_deployment_model_deployment_model_deployment_model_deployment_model_deployment.py` file.
- The `AgentDeploymentModel` class is removed from the `backend/tests/unit/routers/test_agent_deployment_model_deployment_model_deployment_model_deployment_model_deployment_model_deployment_model_deployment_model_deployment_model_deployment_model_deployment_model_deployment_model_deployment.py` file.
- The `AgentDeploymentModel` class is removed from the `backend/tests/unit/routers/test_agent_model_deployment_model_deployment_model_deployment_model_deployment_model_deployment_model_deployment_model_deployment_model_deployment_model_deployment_model_deployment_model_deployment_model_deployment.py` file.
- The `AgentDeploymentModel` class is removed from the `backend/tests/unit/routers/test_agent_deployment_model_deployment_model_deployment_model_deployment_model_deployment_model_deployment_model_deployment_model_deployment_model_deployment_model_deployment_model_deployment_model_deployment_model_deployment.py` file.
- The `AgentDeploymentModel` class is removed from the `backend/tests/unit/routers/test_agent_model_deployment_model_deployment_model_deployment_model_deployment_model_deployment_model_deployment_model_deployment_model_deployment_model_deployment_model_deployment_model_deployment_model_deployment_model_deployment.py` file.
- The `AgentDeploymentModel` class is removed from the `backend/tests/unit/routers/test_agent_deployment_model_deployment_model_deployment_model_deployment_model_deployment_model_deployment_model_deployment_model_deployment_model_deployment_model_deployment_model_deployment_model_deployment_model_deployment_model_deployment.py` file.
- The `AgentDeploymentModel` class is removed from the `backend/tests/unit/routers/test_agent_model_deployment_model_deployment_model_deployment_model_deployment_model_deployment_model_deployment_model_deployment_model_deployment_model_deployment_model_deployment_model_deployment_model_deployment_model_deployment_model_deployment.py` file.
- The `AgentDeploymentModel` class is removed from the `backend/tests/unit/routers/test_agent_deployment_model_deployment_model_deployment_model_deployment_model_deployment_model_deployment_model_deployment_model_deployment_model_deployment_model_deployment_model_deployment_model_deployment_model_deployment_model_deployment_model_deployment.py` file.
- The `AgentDeploymentModel` class is removed from the `backend/tests/unit/routers/test_agent_model_deployment_model_deployment_model_deployment_model_deployment_model_deployment_model_deployment_model_deployment_model_deployment_model_deployment_model_deployment_model_deployment_model_deployment_model_deployment_model_deployment_model_deployment.py` file.
- The `AgentDeploymentModel` class is removed from the `backend/tests/unit/routers/test_agent_deployment_model_deployment_model_deployment_model_deployment_model_deployment_model_deployment_model_deployment_model_deployment_model_deployment_model_deployment_model_deployment_model_deployment_model_deployment_model_deployment_model_deployment_model_deployment.py` file.
- The `AgentDeploymentModel` class is removed from the `backend/tests/unit/routers/test_agent_model_deployment_model_deployment_model_deployment_model_deployment_model_deployment_model_deployment_model_deployment_model_deployment_model_deployment_model_deployment_model_deployment_model_deployment_model_deployment_model_deployment_model_deployment_model_deployment.py` file.
- The `AgentDeploymentModel` class is removed from the `backend/tests/unit/routers/test_agent_deployment_model_deployment_model_deployment_model_deployment_model_deployment_model_deployment_model_deployment_model_deployment_model_deployment_model_deployment_model_deployment_model_deployment_model_deployment_model_deployment_model_deployment_model_deployment_model_deployment.py` file.
- The `AgentDeploymentModel` class is removed from the `backend/tests/unit/routers/test_agent_model_deployment_model_deployment_model_deployment_model_deployment_model_deployment_model_deployment_model_deployment_model_deployment_model_deployment_model_deployment_model_deployment_model_deployment_model_deployment_model_deployment_model_deployment_model_deployment_model_deployment_model_deployment.py` file.
- The `AgentDeploymentModel` class is removed from the `backend/tests/unit/routers/test_agent_deployment_model_deployment_model_deployment_model_deployment_model_deployment_model_deployment_model_deployment_model_deployment_model_deployment_model_deployment_model_deployment_model_deployment_model_deployment_model_deployment_model_deployment_model_deployment_model_deployment_model_deployment_model_deployment.py` file.
- The `AgentDeploymentModel` class is removed from the `backend/tests/unit/routers/test_agent_model_deployment_model_deployment_model_deployment_model_deployment_model_deployment_model_deployment_model_deployment_model_deployment_model_deployment_model_deployment_model_deployment_model_deployment_model_deployment_model_deployment_model_deployment_model_deployment_model_deployment_model_deployment_model_deployment.py` file.
- The `AgentDeploymentModel` class is removed from the `backend/tests/unit/routers/test_agent_deployment_model_deployment_model_deployment_model_deployment_model_deployment_model_deployment_model_deployment_model_deployment_model_deployment_model_deployment_model_deployment_model_deployment_model_deployment_model_deployment_model_deployment_model_deployment_model_deployment_model_deployment_model_deployment_model_deployment.py` file.
- The `AgentDeploymentModel` class is removed from the `backend/tests/unit/routers/test_agent_model_deployment_model_deployment_model_deployment_model_deployment_model_deployment_model_deployment_model_deployment_model_deployment_model_deployment_model_deployment_model_deployment_model_deployment_model_deployment_model_deployment_model_deployment_model_deployment_model_deployment_model_deployment_model_deployment_model_deployment_model_deployment.py` file.
- The `AgentDeploymentModel` class is removed from the `backend/tests/unit/routers/test_agent_deployment_model_deployment_model_deployment_model_deployment_model_deployment_model_deployment_model_deployment_model_deployment_model_deployment_model_deployment_model_deployment_model_deployment_model_deployment_model_deployment_model_deployment_model_deployment_model_deployment_model_deployment_model_deployment_model_deployment_model_deployment_model_deployment.py` file.
- The `AgentDeploymentModel` class is removed from the `backend/tests/unit/routers/test_agent_model_deployment_model_deployment_model_deployment_model_deployment_model_deployment_model_deployment_model_deployment_model_deployment_model_deployment_model_deployment_model_deployment_model_deployment_model_deployment_model_deployment_model_deployment_model_deployment_model_deployment_model_deployment_model_deployment_model_deployment_model_deployment_model_deployment.py` file.
- The `AgentDeploymentModel` class is removed from the `backend/tests/unit/routers/test_agent_deployment_model_deployment_model_deployment_model_deployment_model_deployment_model_deployment_model_deployment_model_deployment_model_deployment_model_deployment_model_deployment_model_deployment_model_deployment_model_deployment_model_deployment_model_deployment_model_deployment_model_deployment_model_deployment_model_deployment_model_deployment_model_deployment_model_deployment.py` file.
- The `AgentDeploymentModel` class is removed from the `backend/tests/unit/routers/test_agent_model_deployment_model_deployment_model_deployment_model_deployment_model_deployment_model_deployment_model_deployment_model_deployment_model_deployment_model_deployment_model_deployment_model_deployment_model_deployment_model_deployment_model_deployment_model_deployment_model_deployment_model_

### CODE REVIEW

File: src/backend/database_models/agent.py [hunk 1]
```diff
@@ -54,116 +29,32 @@ class Agent(Base):
...
-                else None
-            )
-        # TODO Eugene - return the deployment object here when FE is ready Discuss with Scott
-        return (
-            default_model_association.deployment.name
-            if default_model_association
-            else None
-        )
+    def model(self) -> Optional[str]:
+        return self.assigned_model.name if self.assigned_model else None
```
Comment (tianjing-li) [inline src/backend/database_models/agent.py:55]: Nice! A lot cleaner

File: src/backend/routers/agent.py [hunk 2]
```diff
@@ -81,45 +82,34 @@ async def create_agent(
...
+            agent_data = AgentModel(
+                name=agent.name,
+                description=agent.description,
+                preamble=agent.preamble,
+                temperature=agent.temperature,
+                user_id=user_id,
+                organization_id=agent.organization_id,
+                tools=agent.tools,
+                is_private=agent.is_private,
+                deployment_id=deployment_db.id if deployment_db else None,
```
Comment (tianjing-li) [inline src/backend/routers/agent.py:97]: Instead of defaulting to None, can we fallback to Cohere Platform + command-r-plus as defaults?

**Review by tianjing-li — CHANGES_REQUESTED** [review]
Very minor change, then good to go

File: src/backend/database_models/agent.py (hunk 1 above)
Comment (EugeneLightsOn) [reply src/backend/database_models/agent.py:55]: Thank you

File: src/backend/routers/agent.py (hunk 2 above)
Comment (EugeneLightsOn) [reply src/backend/routers/agent.py]: Thank you, I've added defaults for deployment and model.

**Pull request merged**